gluten-flink/runtime/src/main/java/org/apache/gluten/table/runtime/operators/GlutenSingleInputOperator.java

```diff
@@ -109,22 +112,31 @@ public void open() throws Exception {
 
     @Override
     public void processElement(StreamRecord<RowData> element) {
-        inputQueue.add(
-                FlinkRowToVLVectorConvertor.fromRowData(
-                        element.getValue(),
-                        allocator,
-                        session,
-                        inputType));
+        final RowVector inRv = FlinkRowToVLVectorConvertor.fromRowData(
+            element.getValue(),
+            allocator,
+            session,
+            inputType);
+        inputQueue.add(inRv);
         if (result.hasNext()) {
+            final RowVector outRv = result.next();
             List<RowData> rows = FlinkRowToVLVectorConvertor.toRowData(
-                    result.next(),
-                    allocator,
-                    session,
-                    outputType);
+                outRv,
+                allocator,
+                session,
+                outputType);
             for (RowData row : rows) {
                 output.collect(outElement.replace(row));
             }
+            outRv.close();
         }
+      inRv.close();
+    }
+
+    @Override
+    public void close() throws Exception {
+        session.close();
+        memoryManager.close();
```

Comment:
`session.close()` will forcibly close all the associated JNI objects (those who implement Java API `CppObject`). 

`memoryManager.close()` will do a leak check for remaining memory occupations.

But I am not sure it's good to put the code in `#close`.